Sources/_StringProcessing/ByteCodeGen.swift

```diff
@@ -141,9 +142,10 @@ fileprivate extension Compiler.ByteCodeGen {
       }
 
     case .startOfLine:
+      // FIXME: Anchor.startOfLine must always use this first branch
```

Comment:
What does this mean?

For the builtin instruction, I think we'd represent each distinct variation as a separate enum value, so compilation would just be picking the right operand to OR in.

Reply:
It's another regex syntax / DSL syntax discrepancy. Right now, `^` and `Anchor.startOfLine` both get written into the tree as `AST.AssertionKind.startOfLine`, but `^` actually can mean either "start of line" or "start of subject", depending on the current options settings. `Anchor.startOfLine` should always mean what it says, so we need to fix that.

Reply:
We will need a flattened enum post-option-application for the engine. That might also help us understand the model better.